chutney/engine/src/main/java/com/chutneytesting/engine/domain/execution/strategies/RetryWithTimeOutStrategy.java

```diff
@@ -67,15 +69,20 @@ public Status execute(ScenarioExecution scenarioExecution,
             throw new IllegalStateException("Undefined parameter 'retryDelay'"); // TODO - be friendly -> provide a default value instead
         }
 
-        Long timeOutMs = toMilliSeconds(timeOut);
-        Long retryDelayMs = toMilliSeconds(retryDelay);
-        Long timeLeft = timeOutMs;
+
+        Object evaluatedTimeOut = evaluateExpression(timeOut, step.dataEvaluator(), scenarioContext);
+        Object evaluatedRetryDelay = evaluateExpression(retryDelay, step.dataEvaluator(), scenarioContext);
+
+        long timeOutMs = toMilliSeconds(evaluatedTimeOut.toString());
```

Comment:
merge evaluateExpression and toMilliSeconds in one method named, by example : resolveDurationInMilliseconds